x/storage/keeper/msg_server_proofs_test.go

```diff
@@ -0,0 +1,278 @@
+	ditem := h.Sum(nil)
+
+	proof, err := tree.GenerateProof(ditem)
+	if err != nil {
+		return "", "", err
+	}
+
+	jproof, err := json.Marshal(*proof)
+	if err != nil {
+		return "", "", err
+	}
+
+	e := hex.EncodeToString(tree.Root())
+
+	k, _ := hex.DecodeString(e)
+
+	verified, err := merkletree.VerifyProof(ditem, proof, k)
+	if err != nil {
+		fmt.Printf("%v\n", err)
+	}
+
+	if !verified {
+		fmt.Printf("%s\n", "Cannot verify")
+	}
+
+	return fmt.Sprintf("%x", item), string(jproof), nil
+}
+
+func makeContract(file TestFile) (string, string) {
+	f := []byte(file.Data)
+	size := 0
+	var list [][]byte
+
+	size = size + len(f)
+
+	h := sha256.New()
+	io.WriteString(h, fmt.Sprintf("%d%x", 0, f))
+	hashName := h.Sum(nil)
+
+	list = append(list, hashName)
+
+	t, err := merkletree.New(list)
+	if err != nil {
+		fmt.Printf("%v\n", err)
+	}
+
+	return hex.EncodeToString(t.Root()), fmt.Sprintf("%d", size)
+}
+
+func (suite *KeeperTestSuite) TestPostProof() {
+	suite.SetupSuite()
+
+	msgSrvr, keeper, context := setupMsgServer(suite)
+
+	// Create user account
+	user, err := sdk.AccAddressFromBech32("cosmos17j2hkm7n9fz9dpntyj2kxgxy5pthzd289nvlfl")
+	suite.Require().NoError(err)
+
+	// Create provider account
+	testProvider, err := sdk.AccAddressFromBech32("cosmos1ytwr7x4av05ek0tf8z9s4zmvr6w569zsm27dpg")
+	suite.Require().NoError(err)
+
+	// Init Provider
+	_, err = msgSrvr.InitProvider(context, &types.MsgInitProvider{
+		Creator:    testProvider.String(),
+		Ip:         "198.0.0.1",
+		Totalspace: "1_000_000",
+	})
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	// Storage Provider receives file and make merkleroot for contract
+	merkleroot, filesize := makeContract(originalFile)
+
+	// Post Contract
+	_, err = msgSrvr.PostContract(context, &types.MsgPostContract{
+		Creator:  testProvider.String(),
+		Signee:   user.String(),
+		Duration: "1",
+		Filesize: filesize,
+		Fid:      "fid",
+		Merkle:   merkleroot,
+	})
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	// Post Contract #2
+	_, err = msgSrvr.PostContract(context, &types.MsgPostContract{
+		Creator:  testProvider.String(),
+		Signee:   user.String(),
+		Duration: "10",
+		Filesize: "1000",
+		Fid:      "fid2",
+		Merkle:   "invalid_merkleroot",
+	})
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	// Sign Contract for active deal
+	_, err = msgSrvr.SignContract(context, &types.MsgSignContract{
+		Creator: user.String(),
+		Cid:     CID,
+	})
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	// Sign Contract #2 for active deal
+	_, err = msgSrvr.SignContract(context, &types.MsgSignContract{
+		Creator: user.String(),
+		Cid:     CID2,
+	})
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	// Storage Provider get file and create merkle for proof
+	// for tc 1 and 2
+	item, hashlist, err := CreateMerkleForProof(fileFromSP)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	// for tc 3: post proof from a different file
+	item2, hashlist2, err2 := CreateMerkleForProof(randomFile)
+	if err != nil {
+		fmt.Println(err2)
```

Comment:
Can we replace all the `fmt.Println(err)` with a return statement to stop the test from passing?

Reply:
Actually replace with `suite.Require().NoError(err)` instead of the entire if statement

Reply:
got rid of all the fmt.Println(err), fmt.Printf() are cool tho?